minio/minio-js
# Fixed issue with closing files in fPutObject() function

When using the fPutObject() function, in case of an error, the file opened for reading is not closed.

**How to reproduce**

1. Make a test file /tmp/test

```sh
touch /tmp/test
```

2. Create test.js

```js
const Minio = require('minio')

const client = new Minio.Client({
  useSSL: false,
  endPoint: '127.0.0.1',
  port: 9000,
  accessKey: 'minioadmin',
  secretKey: 'minioadmin'
})

setInterval(() => {
  client.fPutObject('test', 'test', '/tmp/test')
    .then(() => console.log('ok'))
    .catch(err => console.log(err))
}, 1000)
```

3. Run test.js

```sh
node test.js
```

4. Check open files for the process

```sh
lsof -p $(pgrep node)

...
node    52282 user   94r      REG                9,0        0 47450575 /tmp/test
node    52282 user   95r      REG                9,0        0 47450575 /tmp/test
node    52282 user   96r      REG                9,0        0 47450575 /tmp/test
```

When files are correctly closed they should not be in this list.
